backend/de.metas.externalsystem/src/main/java/de/metas/externalsystem/export/project/ExportProjectToExternalSystem.java

```diff
@@ -0,0 +1,181 @@
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.externalsystem.export.project;
+
+import ch.qos.logback.classic.Level;
+import de.metas.audit.data.repository.DataExportAuditLogRepository;
+import de.metas.audit.data.repository.DataExportAuditRepository;
+import de.metas.common.externalsystem.JsonExternalSystemName;
+import de.metas.common.externalsystem.JsonExternalSystemRequest;
+import de.metas.common.rest_api.common.JsonMetasfreshId;
+import de.metas.externalsystem.ExternalSystemConfigRepo;
+import de.metas.externalsystem.ExternalSystemConfigService;
+import de.metas.externalsystem.ExternalSystemParentConfig;
+import de.metas.externalsystem.IExternalSystemChildConfig;
+import de.metas.externalsystem.IExternalSystemChildConfigId;
+import de.metas.externalsystem.export.ExportToExternalSystemService;
+import de.metas.externalsystem.rabbitmq.ExternalSystemMessageSender;
+import de.metas.logging.LogManager;
+import de.metas.process.PInstanceId;
+import de.metas.project.Project;
+import de.metas.project.ProjectId;
+import de.metas.project.service.ProjectRepository;
+import de.metas.util.Loggables;
+import de.metas.util.async.Debouncer;
+import lombok.NonNull;
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.util.lang.impl.TableRecordReference;
+import org.compiere.model.I_C_Project;
+import org.slf4j.Logger;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public abstract class ExportProjectToExternalSystem extends ExportToExternalSystemService
+{
+	private static final Logger logger = LogManager.getLogger(ExportProjectToExternalSystem.class);
+
+	@NonNull
+	private final ExternalSystemConfigService externalSystemConfigService;
+	@NonNull
+	private final ProjectRepository projectRepository;
+	@NonNull
+	private final Debouncer<ProjectId> syncProjectDebouncer;
+
+	protected ExportProjectToExternalSystem(
+			@NonNull final ExternalSystemConfigRepo externalSystemConfigRepo,
+			@NonNull final ExternalSystemMessageSender externalSystemMessageSender,
+			@NonNull final DataExportAuditLogRepository dataExportAuditLogRepository,
+			@NonNull final DataExportAuditRepository dataExportAuditRepository,
+			@NonNull final ExternalSystemConfigService externalSystemConfigService,
+			@NonNull final ProjectRepository projectRepository)
+	{
+		super(dataExportAuditRepository, dataExportAuditLogRepository, externalSystemConfigRepo, externalSystemMessageSender);
+
+		this.externalSystemConfigService = externalSystemConfigService;
+		this.projectRepository = projectRepository;
+		this.syncProjectDebouncer = Debouncer.<ProjectId>builder()
+				.name("syncProjectDebouncer")
+				.bufferMaxSize(sysConfigBL.getIntValue("de.metas.externalsystem.debouncer.bufferMaxSize", 100))
```

Comment:
please extract constants for all uses of these two sysconfigs, and for the default values